lib/Dialect/ESI/runtime/cpp/lib/Logging.cpp

```diff
@@ -0,0 +1,98 @@
+//===- Logging.cpp - ESI logging system API implementation ----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// DO NOT EDIT!
+// This file is distributed as part of an ESI package. The source for this file
+// should always be modified within CIRCT (lib/dialect/ESI/runtime/cpp/).
+//
+//===----------------------------------------------------------------------===//
+
+#include "esi/Logging.h"
+#include "esi/Common.h"
+
+#include <iostream>
+#include <mutex>
+
+using namespace esi;
+
+// Necessary in some versions of the c++ standard library to avoid warnings
```

Comment:
@mortbopet I think this is true, but I find it quite unusual... am I doing something wrong?

Reply:
This is the first time i've seen this pattern, so 🤷.